src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -2165,24 +2166,32 @@ VecSimIndexInfo HNSWIndex<DataType, DistType>::info() const {
     return info;
 }
 
+template <typename DataType, typename DistType>
+VecSimIndexStaticInfo HNSWIndex<DataType, DistType>::staticInfo() const {
+    VecSimIndexStaticInfo info = this->getStaticInfo();
```

Comment:
shouldn't be a part of the abstract class?

Reply:
I pit it here since we need to set the backend index algorithm (which is known only here and not in the base class)